homeassistant/helpers/config_validation.py

```diff
@@ -1769,6 +1771,27 @@ def STATE_CONDITION_SCHEMA(value: Any) -> dict[str, Any]:
     )
 )
 
+
+def _backward_compat_trigger_schema(value: Any | None) -> Any:
+    """Backward compatibility for trigger schemas."""
+
+    if not isinstance(value, dict):
+        return value
+
+    # `platform` has been renamed to `trigger` for users
+    # In the code, platform is still used
+    if CONF_TRIGGER in value:
+        if CONF_PLATFORM in value:
+            raise vol.Invalid(
+                "Cannot specify both 'platform' and 'trigger'. Please use 'trigger' only."
+            )
+        response = copy(value)
+        response[CONF_PLATFORM] = response.pop(CONF_TRIGGER)
+        return response
```

Comment:
I'm not sold on naming this `response`, how about this:
```suggestion
        value = copy(value)
        value[CONF_PLATFORM] = value.pop(CONF_TRIGGER)
```